satpy/writers/ninjogeotiff.py

```diff
@@ -418,12 +430,22 @@ def get_date_id(self):
         """Calculate the date ID.
 
         That's seconds since UNIX Epoch for the time corresponding to the
-        satellite image start of measurement time.
+        satellite image reference time or start of measurement time.
         """
         tm = self.dataset.attrs["start_time"]
         delta = tm.replace(tzinfo=datetime.timezone.utc) - self._epoch
         return int(delta.total_seconds())
 
+    def get_valid_date_id(self):
+        """Calculate the valid date ID.
+
+        That's seconds since UNIX Epoch for a representative time for the
+        image.
+        """
+        dt = get_valid_time(self.dataset)
```

Comment:
`dt` is used for `import datetime as dt`, so this is very confusing.

Reply:
Fixed in c3626c430.